src/main/java/bio/terra/cli/businessobject/Workspace.java

```diff
@@ -168,6 +168,19 @@ public void delete() {
     return listedWorkspaces.stream().map(Workspace::new).collect(Collectors.toList());
   }
 
+  /**
+   * Get a resource by name and expect a specific type.
+   *
+   * @throws UserActionableException if the resource is not found or is the wrong type
+   */
+  public <T extends Resource> T getResourceOfType(String name, Resource.Type type) {
```

Comment:
consider making the type casting a method on `Resource` that way it can be used anywhere we're trying to downcast the type instead of only when we're `Workspace.get`.

Reply:
good idea. done.